es-space-mgmt/main/default/lwc/customerTile/__tests__/customerTile.test.js

```diff
@@ -0,0 +1,111 @@
+    }
+
+    it('displays the correct content in the tile', () => {
+        const element = createElement('c-customer-tile', {
+            is: CustomerTile
+        });
+        element.customer = CUSTOMER_DETAILS;
+        element.object = SOBJECT_TYPE;
+        document.body.appendChild(element);
+
+        const linkEl = element.shadowRoot.querySelector('a');
+        expect(linkEl.textContent).toBe(CUSTOMER_DETAILS.name);
+
+        const pElements = element.shadowRoot.querySelectorAll('p');
+        expect(pElements[0].textContent).toBe(
+            `Status: ${CUSTOMER_DETAILS.status}`
+        );
+        expect(pElements[1].textContent).toBe(
+            `Email: ${CUSTOMER_DETAILS.email}`
+        );
+    });
+
+    it('displays the correct icon in the tile', () => {
+        const element = createElement('c-customer-tile', {
+            is: CustomerTile
+        });
+        element.customer = CUSTOMER_DETAILS;
+        element.object = SOBJECT_TYPE;
+        document.body.appendChild(element);
+
+        const lowerCaseObjectName = SOBJECT_TYPE.toLowerCase();
+
+        const iconElement = element.shadowRoot.querySelector('lightning-icon');
+        expect(iconElement.iconName).toBe(`standard:${lowerCaseObjectName}`);
+    });
+
+    it('clicking the tile fires the customerselect custom event', () => {
+        const SELECT_EVENT_DETAIL = {
+            customerId: '0000000',
+            sobjectType: 'Lead',
+            state: 'NY'
+        };
+
+        const element = createElement('c-customer-tile', {
+            is: CustomerTile
+        });
+        element.customer = CUSTOMER_DETAILS;
+        element.object = SOBJECT_TYPE;
+        document.body.appendChild(element);
+
+        // Mock handler for child event
+        const handler = jest.fn();
+        element.addEventListener('customerselect', handler);
+
+        const divEl = element.shadowRoot.querySelector('div');
+        divEl.click();
+
+        return Promise.resolve().then(() => {
+            // Validate if event got fired
+            expect(handler).toHaveBeenCalled();
+            expect(handler.mock.calls[0][0].detail).toEqual(
+                SELECT_EVENT_DETAIL
+            );
+        });
+    });
+
+    it('check if the hyperlink has correct href and alt attributes', () => {
+        const element = createElement('c-customer-tile', {
+            is: CustomerTile
+        });
+        element.customer = CUSTOMER_DETAILS;
+        element.object = SOBJECT_TYPE;
+        document.body.appendChild(element);
+
+        return flushPromises().then(() => {
+            // Validate if event got fired
+            const linkEl = element.shadowRoot.querySelector('a');
+            expect(linkEl.getAttribute('alt')).toBe(
+                `Navigate to ${SOBJECT_TYPE} record detail for ${CUSTOMER_DETAILS.name}`
+            );
+            expect(linkEl.href).toBe(
+                `https://mydomain.my.salesforce.com/${CUSTOMER_DETAILS.Id}`
```

Comment:
Move the static part of the URL as a module constant and add a comment to indicate that it's coming from the NavigationMixin stub.

Reply:
Done